File: gpp_app/lib/domain/model/enumeradores/tipo_asteca_enum.dart
enum TipoAstecaEnum {
  VISTORIA(1),
  REPARO(2);

  final int value;
  const TipoAstecaEnum(this.value);

  String get name {
    switch (this) {
      case TipoAstecaEnum.REPARO:
        return 'Reparo';
      case TipoAstecaEnum.VISTORIA:
        return 'Vistoría';
    }
  }
}

File: gpp-api/src/main/java/com/develop/gpp/domain/enumeradores/TipoAstecaEnum.java
package com.develop.gpp.domain.enumeradores;

import lombok.Getter;
import lombok.ToString;

@Getter
@ToString
public enum TipoAstecaEnum {

    VISTORIA("Vistoria", 1),
    REPARO("Reparo", 2);
  
    private int valor;
    private String descricao;

    private TipoAstecaEnum(String descricao, int valor) {
        this.descricao = descricao;
        this.valor = valor;
    }

}
